graphium/nn/pyg_layers/pooling_pyg.py

```diff
@@ -33,7 +33,7 @@ def scatter_logsum_pool(x: Tensor, batch: LongTensor, dim: int = 0, dim_size: Op
     Returns:
         the pooled features tensor
     """
-    dim_size = int(batch.max().item() + 1) if dim_size is None else dim_size
+    dim_size = int(batch.max().detach() + 1) if dim_size is None else dim_size
```

Comment:
@zhiyil-graphcore can you confirm that the `detach` here won't cause issues on IPUs?

Reply:
I assume pooling_pyg layer is being used by almost all the models? Just tested with toymix and largemix sample dataset, gcn model, it works fine on IPU.